language/move-core/types/src/u256.rs

```diff
@@ -0,0 +1,431 @@
+            U256CastErrorKind::TooLargeForU8 => "u8",
+            U256CastErrorKind::TooLargeForU16 => "u16",
+            U256CastErrorKind::TooLargeForU32 => "u32",
+            U256CastErrorKind::TooLargeForU64 => "u64",
+            U256CastErrorKind::TooLargeForU128 => "u128",
+        };
+        let err_str = format!("Cast failed. {} too large for {}.", self.val, type_str);
+        write!(f, "{err_str}")
+    }
+}
+
+impl std::error::Error for U256FromStrError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        self.0.source()
+    }
+}
+
+impl fmt::Display for U256FromStrError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Copy, PartialOrd, Ord)]
+pub struct U256(PrimitiveU256);
+
+impl fmt::Display for U256 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl std::str::FromStr for U256 {
+    type Err = U256FromStrError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::from_str_radix(s, 10)
+    }
+}
+
+impl<'de> Deserialize<'de> for U256 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(U256::from_le_bytes(
+            &(<[u8; U256_NUM_BYTES]>::deserialize(deserializer)?),
+        ))
+    }
+}
+
+impl Serialize for U256 {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.to_le_bytes().serialize(serializer)
+    }
+}
+
+impl Shl<u8> for U256 {
+    type Output = Self;
+
+    fn shl(self, rhs: u8) -> Self::Output {
+        let Self(lhs) = self;
+        Self(lhs << rhs)
+    }
+}
+
+impl Shr<u8> for U256 {
+    type Output = Self;
+
+    fn shr(self, rhs: u8) -> Self::Output {
+        let Self(lhs) = self;
+        Self(lhs >> rhs)
+    }
+}
+
+impl BitOr<U256> for U256 {
+    type Output = Self;
+
+    fn bitor(self, rhs: U256) -> Self::Output {
+        let Self(lhs) = self;
+        let Self(rhs) = rhs;
+        Self(lhs | rhs)
+    }
+}
+
+impl BitAnd<U256> for U256 {
+    type Output = Self;
+
+    fn bitand(self, rhs: U256) -> Self::Output {
+        let Self(lhs) = self;
+        let Self(rhs) = rhs;
+        Self(lhs & rhs)
+    }
+}
+
+impl BitXor<U256> for U256 {
+    type Output = Self;
+
+    fn bitxor(self, rhs: U256) -> Self::Output {
+        let Self(lhs) = self;
+        let Self(rhs) = rhs;
+        Self(lhs ^ rhs)
+    }
+}
+
+impl U256 {
+    /// Zero value as U256
+    pub const fn zero() -> Self {
+        Self(PrimitiveU256::zero())
+    }
+
+    /// One value as U256
+    pub const fn one() -> Self {
+        Self(PrimitiveU256::one())
+    }
+
+    /// Max value of U256: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+    pub const fn max() -> Self {
+        Self(PrimitiveU256::max_value())
+    }
```

Comment:
Please rename this to `max_value` so to be consistent with the primitive integer types. Similarly we may also want to add an associated constant MAX.

By convention `max` is used for getting the bigger number from two. See https://doc.rust-lang.org/std/primitive.u64.html#impl-Ord-for-u64 for example